server/src/test/java/org/elasticsearch/health/HealthServiceTests.java

```diff
@@ -94,30 +112,6 @@ public void onFailure(Exception e) {
         };
     }
 
-    public void testDuplicateIndicatorNames() throws Exception {
```

Comment:
Removed this test as I don’t think we should test assertions (if they’re not assertions they should be failures) and I don’t think we should surface assertions in the async listener.

Reply:
Is this related to this PR? I don't think I see the connection.

Reply:
This assertion is now in a background thread https://github.com/elastic/elasticsearch/pull/91628/files#diff-444fdb9629cdf9a3ece311d4e8ecd627777532a0f878afc2c7aac5cc3e8ebd6aR190

Irrespective of that, I don't think we should be testing the assertions (if it's not an invariant I think it should be a failure and that failure should be tested via the listener.onFailure thing). I don't think that's the case here though.

Reply:
OK I see the reason for removing it -- it's an Error in the background thread, and we only catch Exceptions. So we're going to lose test coverage for `findDuplicatesByName()`. But I guess that's not a huge loss.